### PR TITLE
[DNM] Delete PipelineRuns immediately after completion

### PR DESCRIPTION
**Pull request by Roming22**
rh-pre-commit.version: 2.0.3
rh-pre-commit.check-secrets: ENABLED

### CODE REVIEW

File: operator/gitops/argocd/pipeline-service/tekton-results/watcher-config.yaml [hunk 1]
```diff
@@ -19,7 +19,7 @@ spec:
               "token",
               "-check_owner=false",
               "-completed_run_grace_period",
-              "10m",
+              "-1m",
```
Comment (enarha) [inline operator/gitops/argocd/pipeline-service/tekton-results/watcher-config.yaml:22]: I would make that just "-1". It's less confusing than -1 minute.

Comment (enarha) [reply operator/gitops/argocd/pipeline-service/tekton-results/watcher-config.yaml:22]: Please ignore. That actually results an error.

**Comment by gabemontero** [comment]
Per discussion in slack, some of the upstream metrics, and all of the pipeline service exporter metrics, are accumulated after a pipelinerun is completed.

Odd timing issues, race conditions, could arise as if the deletes happen this fast.

Testing / more investigation is most likely needed.  Possibly some sort of lessening of the time out, but not going immediately, might be needed.

**Comment by ramessesii2** [comment]
We've a duplicate PR #868. I'll close that in favor of this one. However, as [this comment](https://github.com/openshift-pipelines/pipeline-service/pull/868#discussion_r1407526875) on #868 mentions that we get into race conditions with immediate pruning, and there is a downstream JIRA issue dedicated to fixing it which is most likely needed to get this PR merged.

**Comment by Roming22** [comment]
Do not merge this PR until the issue with logs not being synced before the resource is deleted is solved.

**Comment by gabemontero** [comment]
> Do not merge this PR until the issue with logs not being synced before the resource is deleted is solved.

Add to ^^:  until we do sufficient testing/verification that knative/controller-runtime won't aggregate events such that our various metrics, both upstream and the exporter, that are calculated after the pipelinerun is complete are not negatively impacted

**Comment by Roming22** [comment]
@gabemontero @enarha I'll let you decide what you want to do with this PR. You may decide you do not want to delete PLR as soon as they complete.

**Comment by gabemontero** [comment]
> @gabemontero @enarha I'll let you decide what you want to do with this PR. You may decide you do not want to delete PLR as soon as they complete.

yeah we are a ways from being able to do this IMO

**Pull request closed without merging**